test/controllers/ImporterAddressFrontendControllerSpec.scala

```diff
@@ -17,57 +17,66 @@
 package controllers
 
 import base.SpecBase
+import data.TestData
 import forms.ImporterManualAddressFormProvider
-import models.{Address, UserAnswers}
+import models.{Address, Country, UserAnswers}
 import org.mockito.Matchers.any
 import org.mockito.Mockito.when
 import org.scalatestplus.mockito.MockitoSugar
-import pages.ImporterManualAddressPage
+import pages.ImporterAddressPage
+import play.api.inject.bind
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
+import services.CountryService
 import uk.gov.hmrc.govukfrontend.views.Aliases.SelectItem
 import views.html.ImporterManualAddressView
 
 import scala.concurrent.Future
 
-class ImporterManualAddressControllerSpec extends SpecBase with MockitoSugar {
+class ImporterAddressFrontendControllerSpec extends SpecBase with MockitoSugar {
 
-  val formProvider = new ImporterManualAddressFormProvider()
-  val form         = formProvider()
+  implicit val countriesService = TestData.testCountryService
+  val formProvider              = new ImporterManualAddressFormProvider()
+  val form                      = formProvider()
 
-  lazy val importerManualAddressRoute = routes.ImporterManualAddressController.onPageLoad().url
+  lazy val importerManualAddressRoute = routes.ImporterAddressFrontendController.onPageLoad().url
 
-  "ImporterManualAddress Controller" must {
+  "ImporterAddressFrontendController" must {
 
-    "return OK and the correct view for a GET" in {
+    "redirects to address lookup when there is no address in cache" in {
 
-      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers)).build()
+      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers))
+        .overrides(bind[CountryService].toInstance(countriesService))
+        .build()
 
       val request = FakeRequest(GET, importerManualAddressRoute)
 
       val result = route(application, request).value
 
-      val view = application.injector.instanceOf[ImporterManualAddressView]
-
-      status(result) mustEqual OK
+      status(result) mustEqual SEE_OTHER
 
-      contentAsString(result) mustEqual
-        view(form, defaultBackLink, false, Seq(SelectItem(text = "United Kingdom", value = Some("GB"))))(
-          request,
-          messages
-        ).toString
+      redirectLocation(result).value mustEqual routes.ImporterAddressFrontendController.onChange().url
 
       application.stop()
     }
 
     "populate the view correctly on a GET when the question has previously been answered" in {
 
       val userAnswers = UserAnswers(userAnswersId).set(
-        ImporterManualAddressPage,
-        Address("address line 1", Some("address line 2"), "city", Some("Region"), "GB", "AA211AA")
+        ImporterAddressPage,
```

Comment:
AMBER: Ditto from above, but for the importer controller. "We should also add and test an international address (without a postcode, as above comments). I'm not expecting the controller to return validation failures for them."